src/Modal.js

```diff
@@ -277,6 +277,11 @@ const Modal = React.createClass({
     }
   },
 
+  componentWillUnmount() {
+    this.onHide();
+    this.props.onHide();
```

Comment:
sorry, that was a probably a bit unclear. 

If a users clicks something to hide the modal, eg the backdrop, props.onHide is called. The parent component handles the callback and sets some state which causes the modal to not be rendered, which results in a call to componentWillUnmount, which then calles props.onHide again.